packages/domain-picker/src/domain-categories/index.tsx

```diff
@@ -6,28 +6,27 @@ import { Button, Icon } from '@wordpress/components';
 import classNames from 'classnames';
 import { useI18n } from '@automattic/react-i18n';
 import { useState } from '@wordpress/element';
-import { useSelect } from '@wordpress/data';
-
-/**
- * Internal dependencies
- */
-import { DOMAIN_SUGGESTIONS_STORE } from '../../stores/domain-suggestions';
+import { DomainSuggestions } from '@automattic/data-stores';
 
 /**
  * Style dependencies
  */
 import './style.scss';
 
+type DomainCategory = DomainSuggestions.DomainCategory;
+
 export interface Props {
 	onSelect: ( domainCategorySlug?: string ) => void;
 	selected?: string;
+	domainCategories: DomainCategory[];
 }
 
-const DomainPickerCategories: React.FunctionComponent< Props > = ( { onSelect, selected } ) => {
+const DomainPickerCategories: React.FunctionComponent< Props > = ( {
+	onSelect,
+	selected,
+	domainCategories,
+} ) => {
 	const { __ } = useI18n();
-	const domainCategories = useSelect( ( select ) =>
-		select( DOMAIN_SUGGESTIONS_STORE ).getCategories()
-	);
```

Comment:
We chatted on the call, but this "simple" fetching can continue to remain in the component. I'm OK with using data store's string key directly `automattic/domains/suggestions`.